pkg/apiserver/internal/v1/account.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2020 The KubeCarrier Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	catalogv1alpha1 "github.com/kubermatic/kubecarrier/pkg/apis/catalog/v1alpha1"
+	v1 "github.com/kubermatic/kubecarrier/pkg/apiserver/api/v1"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/oidc"
+	"github.com/kubermatic/kubecarrier/pkg/apiserver/internal/util"
+)
+
+type accountServer struct {
+	client client.Client
+}
+
+var _ v1.AccountServiceServer = (*accountServer)(nil)
+
+// +kubebuilder:rbac:groups=catalog.kubecarrier.io,resources=accounts,verbs=get;list;watch
+
+func NewAccountServiceServer(c client.Client) v1.AccountServiceServer {
+	return &accountServer{
+		client: c,
+	}
+}
+
+func (o accountServer) List(ctx context.Context, req *v1.AccountListRequest) (res *v1.AccountList, err error) {
+	user, present := oidc.ExtractUserInfo(ctx)
```

Comment:
As a side note, maybe when the user isn't authenticated, we could inject the `system:anonymous` user by default. 

Right now all endpoints are under OIDC middleware, and must pass auth procedure. This would allow more granular permissions, for example, the version endpoint might as well be available publically. 

https://kubernetes.io/docs/reference/access-authn-authz/authentication/#anonymous-requests

Reply:
Yeah, that's a good catch, but do you actually want anonymous users to list accounts?

Reply:
Nope; my thought process is you shouldn't handle auth explicitly in every API call. ( `if !present ..`). It should fail naturally when you try using the client with the `system:anonymous` user.

It seems kinda silly both having an authentication middleware and you still have to check in every endpoint explicitly is the user authenticated.

Reply:
This looks like we should put this into a grpc middleware and just extract the authenticated userinfo from the context for authorization.

Maybe we can use:
https://github.com/grpc-ecosystem/go-grpc-middleware
(This adds even more capability for tracing and monitoring)

Seems like you can even specify advanced validation:
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/validator/doc.go

Issues: https://github.com/kubermatic/kubecarrier/issues/441

Reply:
This is for further improvement right? Not blocking this PR merging